packages/components/addon/components/hds/gutters/index.js

```diff
@@ -0,0 +1,99 @@
+import Component from '@glimmer/component';
+import { htmlSafe } from '@ember/template';
+import { assert } from '@ember/debug';
+
+export const DIRECTIONS = ['horizontal', 'vertical'];
+export const SPACING_SIZES = ['xxs', 'xs', 'sm', 'md', 'lg', 'xl', 'xxl'];
+
+const regEx = /^((\d+)|(\d+\.\d+)|(\.\d+))(px|rem|em|%)$/;
+
+export default class HdsGuttersIndexComponent extends Component {
+  /**
+   * Get the direction for the component
+   * Accepted values: 'horizontal', 'vertical'
+   * @param direction
+   * @type {string}
+   * @default ? // TODO: Determine if a default should be set & if so which value
+   */
+  get direction() {
+    let { direction } = this.args;
+
+    assert(
+      '@direction for "Hds::Gutters" must have a valid value',
+      DIRECTIONS.includes(direction)
+    );
+
+    return direction;
+  }
+
+  /**
+   * Get the preset spacing size to apply to the item.
+   * Accepted values: 'xxs', 'xs', 'sm', 'md', 'lg', 'xl', 'xxl'
```

Comment:
To be discussed if we want to re-introduce a spacing scale (time ago the decision was not to, and I would like to see strong arguments to justify this change).

Reply:
### Background behind my original thinking:

Currently within the old Structure library and within Cloud-ui code, they are using spacing scales following very similar naming: "xs", "sm", "md", etc. so this type of naming isn't a new or unfamiliar concept to devs.

The spacing scale we have in Figma uses numbers as names instead of letters but we aren't currently referencing it directly anywhere; it doesn't get exported as tokens. So we could easily add naming such as "xs", "sm" etc. to the Figma file in addition to the numbers or instead of them if we are worried about potential confusion.

### Trying to use px numbers for spacing size names instead...

Currently, for users to use one of the preset spacing sizes with the Gutters component, they just need to add the short abbreviation name.

**Example:**  
```
<Hds::Gutters @direction="horizontal" @spacing="xs">
	<span>item 1</span>
	<span>item 2</span>
	<span>item 3</span>
</Hds::Gutters>
```

If I change the spacing size names to use numbers based on px instead, then users would enter a plain number.

**Example:**  
`<Hds::Gutters @direction="horizontal" @spacing="4" />`

It then becomes confusing what "4" means. Does it mean a pixel value? Or does it mean size #4 in the HDS Figma file which is actually 16px?

To try to prevent this confusion, we could include "px" as part of the spacing names.

**Example:**  
`<Hds::Gutters @direction="horizontal" @spacing="4px" />`

But then there is no difference between using a preset size vs. entering a custom CSS unit size.

So then we have to think of other ways to solve the confusion. Do we require users to enter the full variable names? We could, but then the component becomes harder to use and users need to type more, have more chance of typing the wrong name, or have to constantly look up the names for the spacing scale.

**Example:**  
`<Hds::Gutters @direction="horizontal" @spacing="hds-spacing-4px" />`

To me it seems that using short descriptive abbreviations that devs are already familiar with would make the components simpler and easier to use. Please let me know if you have other ideas I haven't considered though.

Reply:
@didoo Do you have any comments or suggestions? I tried to figure out how to follow what you wanted but it seems like it introduces problems and confusion of its own (at least in my thinking). Do you have any ideas on what would make sense?

Reply:
> So we could easily add naming such as "xs", "sm" etc. to the Figma file in addition to the numbers or instead of them if we are worried about potential confusion.

True. But this still would not solve the problem of: _"As a developer, I am inspecting a Figma file with the design specs. I see that the spacing between two elements is 4px. Which size alias should I use? xs? sm? other?"_ The only way is 1) I know it by heart or 2) I have to look out for this info somewhere.

> It then becomes confusing what "4" means. Does it mean a pixel value? Or does it mean size https://github.com/hashicorp/design-system/pull/4 in the HDS Figma file which is actually 16px?

No one is using the size scale in Figma. Not even the designers. We could remove that scale and probably no one would notice. So I don't think there would be confusion, or at least would be much less than with any alternative approach we could use.

> To try to prevent this confusion, we could include "px" as part of the spacing names.
> But then there is no difference between using a preset size vs. entering a custom CSS unit size.

**Exactly** the point I wanted to make during our discussion! Given these "boundary conditions", why do you need a "scale"? Just let the users enter the spacing as they want (including `px`) and you reduce **by a lot** the overall complexity of this component.

All the complexity is rising because of the introduction of an artificial spacing scale that acts as a proxy between the user and the actual value they need. As soon as one removes that scale, and the proxy/mapping, then it's immediate the relation between what they see in Figma (`4px`), the value they need to use in code (`4px`), the value that will be used in the component (`4px`) and the value that will be rendered in the browser (`4px`; until we use a post-css processor that converts this to `rem`s, but this is for the future).

I know this approach is an anti-pattern. It took me quite a while to accept it too, but once you get used to it you see that makes sense: we haven't had any issue since we decided to follow this approach almost one year ago.

Reply:
@meirish commented on my CRD that the blocking issue at a past attempt to create a layout component was the lack of a spacing scale.

Reply:
He doesn't remember correctly: the blocking issue was lack of time for all of us to continue (we were supposed to continue working on that PR/draft together, but we never did).

Reply:
Ok. It would be nice to start to establish some some of consistent values for spacing within layouts still to my thinking. Also I'd like to at some point be able to switch to using rem units for spacing sizes instead of pixels. I don't know how realistic either of those are though...

Reply:
> I'd like to at some point be able to switch to using rem units for spacing sizes instead of pixels. 

As mentioned before, this will be done automatically (in the future) using a CSS post-processor. Sizes will still be declared in `px` in the Sass source code.

Reply:
Ok